presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -0,0 +1,269 @@
+ * support adding more splits during execution.
+ */
+public class NativeExecutionTask
+{
+    private static final Logger log = Logger.get(NativeExecutionTask.class);
+    private static final int GET_TASK_INFO_INTERVALS = 100;
+    private final Session session;
+    private final TaskId taskId;
+    private final PlanFragment planFragment;
+    @GuardedBy("this")
+    private final AtomicReference<OutputBuffers> outputBuffers = new AtomicReference<>();
+    private final HttpClient httpClient;
+    private final TableWriteInfo tableWriteInfo;
+    private final RemoteTaskStats stats;
+    private final List<TaskSource> sources;
+    private TaskInfo taskInfo;
+    private final Executor executor;
+    private final ScheduledExecutorService updateScheduledExecutor;
+    @GuardedBy("this")
+    private ScheduledFuture<?> scheduledFuture;
+    private final JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec;
+    private final JsonCodec<PlanFragment> planFragmentCodec;
+    private final JsonCodec<TaskInfo> taskInfoCodec;
+
+    public NativeExecutionTask(
+            Session session,
+            TaskId taskId,
+            String nodeId,
+            URI location,
+            PlanFragment planFragment,
+            List<TaskSource> sources,
+            OutputBuffers outputBuffers,
+            HttpClient httpClient,
+            TableWriteInfo tableWriteInfo,
+            RemoteTaskStats stats,
+            Executor executor,
+            ScheduledExecutorService updateScheduledExecutor,
+            JsonCodec<TaskUpdateRequest> taskUpdateRequestCodec,
+            JsonCodec<PlanFragment> planFragmentCodec,
+            JsonCodec<TaskInfo> taskInfoCodec)
+    {
+        requireNonNull(session, "session is null");
+        requireNonNull(taskId, "taskId is null");
+        requireNonNull(nodeId, "nodeId is null");
+        requireNonNull(location, "location is null");
+        requireNonNull(planFragment, "planFragment is null");
+        requireNonNull(sources, "sources is null");
+        requireNonNull(outputBuffers, "outputBuffers is null");
+        requireNonNull(httpClient, "httpClient is null");
+
+        try (SetThreadName ignored = new SetThreadName("HttpRemoteTask-%s", taskId)) {
+            this.taskId = taskId;
+            this.session = session;
+            this.planFragment = planFragment;
+            this.outputBuffers.set(outputBuffers);
+            this.httpClient = httpClient;
+            this.tableWriteInfo = tableWriteInfo;
+            this.stats = stats;
+            this.sources = sources;
+            this.executor = executor;
+            this.updateScheduledExecutor = updateScheduledExecutor;
+            List<BufferInfo> bufferStates = outputBuffers.getBuffers()
+                    .keySet().stream()
+                    .map(outputId -> new BufferInfo(outputId, false, 0, 0, PageBufferInfo.empty()))
+                    .collect(toImmutableList());
```

Comment:
Some coding style issue @highker can help if I'm wrong:
Can we have one method call per line here?

Reply:
The current code style is correct. Check https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting

`toImmutableList()` is a subprocedure of `collect`; so it's correct this way

Reply:
@highker  good to know. Thank James